jscomp/others/jsx.ml

```diff
@@ -37,3 +37,7 @@ type 'props component = ('props, element) componentLike
 (* this function exists to prepare for making `component` abstract *)
 external component : ('props, element) componentLike -> 'props component
   = "%identity"
+
+let addKeyProp (o : 't) k =
+  Obj.magic (Js.Obj.assign (Obj.magic o) [%obj { key = k }])
+  [@@inline]
```

Comment:
Not sure whether `[@@inline]` is correct or not. I'm wondering `[@@bs.inline]` is current, but it fails to be compiled. `Unused attribute ...` error is thrown.

Reply:
Js output seems incorrect either.
```js
// generated
Jsx.addKeyProp({ x: "x" }, { key: "k1"})

// expected
Object.assign({ x: "x" }, { key: "k1"})
```

Reply:
Same:
```ml
let is_inline : attr -> bool =
 fun ({ txt }, _) -> txt = "bs.inline" || txt = "inline"
```

I think it's more like inline is not so well specified. Not even sure it works across files.

Reply:
I would assume a bundler would inline it anyway?

Reply:
> I would assume a bundler would inline it anyway?

Worth checking. The current output is working fine without any issues in the example project anyway.

Reply:
OK to merge?